scan-repo-for-secrets.sh

```
@@ -0,0 +1,20 @@
+if [ -z "$REPO_PATH" ]; then
+    echo "Usage: $0 /path/to/repository"
+    exit 1
+fi
+
+if [ ! -d "$REPO_PATH/.git" ]; then
+    echo "Not a git repository: $REPO_PATH"
+    exit 1
+fi
+
+echo "Scanning $REPO_PATH for secrets in history..."
+cd "$REPO_PATH"
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)